### PR TITLE
feat: add csrf-with-session example

### PR DESCRIPTION
**Pull request by sixcolors**
Adds example for csrf with session.

### CODE REVIEW

File: csrf-with-session/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+### Submit the form on the /protected page
+
+Once logged in, you will be able to see the /protected page. The /protected page contains a form that submits to the /protected page. If you try to submit the form without a valid CSRF token, you will get a 403 Forbidden error.
+
+## CSRF Protection
+
+All methods except GET, HEAD, OPTIONS, and TRACE are checked for the CSRF token. If the token is not present or does not match the token in the session, the request is aborted with a 403 Forbidden error.
+
+## Token Lyfecycle
```
Comment (gaby) [inline csrf-with-session/README.md:79]: Typo here: Lifecycle

File: csrf-with-session/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+The CSRF token is generated when the user visits any page on the site. The token is stored in the session and is valid for until it expires, or the authorization scope changes (e.g. the user logs in, or logs out).
+
+It is important that CSRF tokens do not presist beyond the scope of the user's session, that a new session is created when the user logs in, and that the session is destroyed when the user logs out.
```
Comment (gaby) [inline csrf-with-session/README.md:83]: Persist

**Review by gaby — CHANGES_REQUESTED** [review]

File: csrf-with-session/README.md (hunk 1 above)
Comment (sixcolors) [reply csrf-with-session/README.md]: fixed, thx

File: csrf-with-session/README.md (hunk 2 above)
Comment (sixcolors) [reply csrf-with-session/README.md]: fixed, thx again :)

**Review by gaby — APPROVED** [review]

**Pull request merged**